src/components/molecules/AnnouncementMessage/AnnouncementMessage.tsx

```diff
@@ -1,51 +1,89 @@
+
+import { LinkButton } from "components/atoms/LinkButton";
 
 import "./AnnouncementMessage.scss";
 
-type AnnouncementMessageProps = {
-  message?: string;
-};
+export interface AnnouncementMessageProps {
+  banner?: Banner;
+  announcementForUser?: boolean;
+}
 
 export const AnnouncementMessage: React.FC<AnnouncementMessageProps> = ({
-  message = "",
+  banner,
+  announcementForUser = false,
 }) => {
-  const [isVisible, setVisibility] = useState<boolean>(false);
-  const { isExpanded } = useChatSidebarControls();
-
-  const hideAnnouncement = useCallback(() => {
-    setVisibility(false);
-  }, []);
+  const {
+    isShown: isShownAnnouncementMessage,
+    show: showAnnouncementMessage,
+    hide: hideAnnouncementMessage,
+  } = useShowHide();
 
   useEffect(() => {
-    if (message) {
-      setVisibility(true);
+    if (isDefined(banner?.content)) {
+      showAnnouncementMessage();
     }
-  }, [message]);
+  }, [banner, showAnnouncementMessage]);
 
-  if (!isVisible || !message) return null;
+  const noop = (e: React.MouseEvent<HTMLElement>) => {
+    e.stopPropagation();
+  };
+
+  const isActiveButton =
+    banner?.buttonDisplayText && banner?.buttonUrl && banner?.isActionButton;
+
+  const containerClasses = classNames("AnnouncementMessage__container", {
+    "AnnouncementMessage__container--centered": banner?.isFullScreen,
+    "AnnouncementContainer--canceled": !announcementForUser,
+  });
+
+  if (!announcementForUser && !banner?.content)
+    return (
+      <div className="AnnouncementMessage">
+        <span className="AnnouncementMessage__default-text">
+          No announcement
+        </span>
+      </div>
+    );
+
+  if (!isShownAnnouncementMessage) return null;
 
   return (
-    <div
-      className={classNames("announcement-container", {
-        centered: !isExpanded,
-      })}
-    >
-      <div className="announcement-message">{getLinkFromText(message)}</div>
-      <span className="close-button" onClick={hideAnnouncement}>
-        <FontAwesomeIcon icon={faTimesCircle} />
-      </span>
+    <div className={containerClasses} onClick={hideAnnouncementMessage}>
+      <div className="AnnouncementMessage" onClick={noop}>
+        {banner?.title && (
+          <h2 className="AnnouncementMessage__title">{banner.title}</h2>
+        )}
+        <div className="AnnouncementMessage__content">
+          {getLinkFromText(banner.content)}
+        </div>
+        {isActiveButton && banner.buttonUrl && (
+          <LinkButton
+            href={banner.buttonUrl}
+            className="AnnouncementMessage__action-button"
+          >
+            {banner.buttonDisplayText}
+          </LinkButton>
+        )}
+        {announcementForUser && banner.hasCloseButton ? (
+          <span
+            className="AnnouncementMessage__close-button"
+            onClick={hideAnnouncementMessage}
+          >
+            <FontAwesomeIcon icon={faTimesCircle} />
+          </span>
+        ) : null}
+      </div>
     </div>
```

Comment:
**Please** add a blank line between sibling blocks here. This is so convolutedly 'smashed together' here that my brain can't even make sense of what is going on. Needs more spacing!

Reply:
There is also a lot of multiple boolean checks going on in the JSX. They should be extracted into `const`s so that there is just 1 check 1 action. Not multiple checks for a single action.

Reply:
The ternary or `null` for `announcementForUser && banner.hasCloseButton` is pointless. Just use a standard `&&` conditional render.

Reply:
There may be more needed here, but it's hurting my brain to try and parse it at the moment.

Reply:
done 771b70a